invoice/src/main/java/org/killbill/billing/invoice/api/user/DefaultInvoiceUserApi.java

```diff
@@ -347,9 +348,15 @@ public InvoiceItem insertCreditForInvoice(final UUID accountId, final UUID invoi
                 // Create an invoice for that credit if it doesn't exist
                 final Invoice invoiceForCredit;
                 if (invoiceId == null) {
-                    invoiceForCredit = new DefaultInvoice(accountId, effectiveDate, effectiveDate, currency);
+                    invoiceForCredit = new DefaultInvoice(accountId, effectiveDate, effectiveDate, currency, InvoiceStatus.DRAFT);
                 } else {
                     invoiceForCredit = getInvoiceAndCheckCurrency(invoiceId, currency, context);
+                    // TODO check with @sbrossie if really want to add this validation
```

Comment:
> We indeed want this check.

There are two test that start failing now with this check.
* [TestAdjustFullInvoice](https://github.com/killbill/killbill/blob/master/invoice/src/test/java/org/killbill/billing/invoice/api/user/TestDefaultInvoiceUserApi.java#L201-L202)
* [TestAdjustPartialInvoice](https://github.com/killbill/killbill/blob/master/invoice/src/test/java/org/killbill/billing/invoice/api/user/TestDefaultInvoiceUserApi.java#L231-L232)

What do you want to do on those tests? Is this the expected behaviour now? (Tests can expect an Exception)